akka-http/src/main/scala/akka/http/scaladsl/server/directives/BasicDirectives.scala

```diff
@@ -301,6 +305,13 @@ trait BasicDirectives {
    * @group basic
    */
   def extractDataBytes: Directive1[Source[ByteString, Any]] = BasicDirectives._extractDataBytes
+
+  /**
+   * Extracts the [[akka.http.scaladsl.model.HttpEntity.Strict]] from the [[akka.http.scaladsl.server.RequestContext]].
+   *
+   * @group basic
+   */
+  def extractStrictEntity(timeout: FiniteDuration): Directive1[HttpEntity.Strict] = BasicDirectives._extractStrictEntity(timeout)
```

Comment:
Please add a warning that this will read the entire request entity into memory regardless of size and also a doc entry about the timeout parameter.